cpp/src/io/csv/csv_reader.cu

```diff
@@ -394,36 +355,29 @@ gdf_error read_csv(csv_read_arg *args)
 	assert(h_uncomp_data != nullptr);
 	assert(h_uncomp_size != 0);
 
-	error = launch_countRecords(h_uncomp_data, h_uncomp_size, raw_csv->opts.terminator, raw_csv->opts.quotechar, raw_csv->num_records);
-	if (error != GDF_SUCCESS) {
-			return error;
-	}
-
-	const bool include_first_row = args->byte_range_offset==0;
-	if (!include_first_row) {
-		--raw_csv->num_records;
-	}
+	error = launch_countRecords(h_uncomp_data, h_uncomp_size, raw_csv, raw_csv->num_records);
+	checkError(error, "call to record number of rows");
 
 	//-----------------------------------------------------------------------------
 	//-- Allocate space to hold the record starting point
-	RMM_TRY( RMM_ALLOC((void**)&(raw_csv->recStart), (sizeof(cu_recstart_t) * (raw_csv->num_records + 1)), 0) ); 
+	RMM_TRY( RMM_ALLOC(&raw_csv->recStart, sizeof(cu_recstart_t) * raw_csv->num_records, 0) ); 
 
 	//-----------------------------------------------------------------------------
 	//-- Scan data and set the starting positions
-	error = launch_storeRecordStart(h_uncomp_data, h_uncomp_size, include_first_row, raw_csv);
+	error = launch_storeRecordStart(h_uncomp_data, h_uncomp_size, raw_csv);
 	checkError(error, "call to record initial position store");
 
 	// Previous kernel stores the record pinput_file.typeositions as encountered by all threads
 	// Sort the record positions as subsequent processing may require filtering
 	// certain rows or other processing on specific records
-	thrust::sort(thrust::device, raw_csv->recStart, raw_csv->recStart + raw_csv->num_records + 1);
+	thrust::sort(thrust::device, raw_csv->recStart, raw_csv->recStart + raw_csv->num_records);
 
```

Comment:
Please use `rmm::exec_policy(stream)` to avoid thrust calling cudaMalloc for sort internal intermediate results

Reply:
Done! Thanks!